### PR TITLE
Log cleanup for velero-plugin-for-vsphere

### PR DESCRIPTION
**Pull request by xinyanw409**
Do logs clean up for velero-plugin-for-vsphere:

Adjust log level
Adjust log message
Log list table: https://confluence.eng.vmware.com/display/~wxinyan/Logs+and+Log+Level+for+Astrolabe+and+Velero-plugin-for-vsphere

Test done:
Command "make" can pass.
CI/CD pipeline build succeed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,7 +52,7 @@ platform_temp = $(subst -, ,$(ARCH))
 GOOS = $(word 1, $(platform_temp))
 GOARCH = $(word 2, $(platform_temp))
 
-BUILDER_IMAGE := golang:1.14
+BUILDER_IMAGE := golang:1.13
```
Comment (dsu-igeek) [inline Makefile]: This shouldn't be a change in this PR, it's already in master.  Did you rebase first?

File: pkg/cmd/cli/download/delete.go [hunk 2]
```diff
@@ -21,6 +22,7 @@ func NewDeleteCommand(f client.Factory, use string) *cobra.Command {
 		Run: func(c *cobra.Command, args []string) {
 			logLevel := logLevelFlag.Parse()
 			logger := logging.DefaultLogger(logLevel, formatFlag.Parse())
+			logger.Debugf("setting log-level to %s", strings.ToUpper(logLevel.String()))
 			logger.Infof("The command, datamgr download delete, is called")
```
Comment (dsu-igeek) [inline pkg/cmd/cli/download/delete.go]: Let's change "The command, datamgr download delete, is called " to "Datamgr download delete called"

File: pkg/cmd/cli/upload/create.go [hunk 3]
```diff
@@ -27,6 +28,7 @@ func NewCreateCommand(f client.Factory, use string) *cobra.Command {
 			//cmd.CheckError(o.Run(c, f))
 			logLevel := logLevelFlag.Parse()
 			logger := logging.DefaultLogger(logLevel, formatFlag.Parse())
+			logger.Debugf("setting log-level to %s", strings.ToUpper(logLevel.String()))
 			logger.Infof("Starting datamgmt server")
```
Comment (dsu-igeek) [inline pkg/cmd/cli/upload/create.go]: datamgmt -> datamgr

File: pkg/cmd/cli/upload/delete.go [hunk 4]
```diff
@@ -21,6 +22,7 @@ func NewDeleteCommand(f client.Factory, use string) *cobra.Command {
 		Run: func(c *cobra.Command, args []string) {
 			logLevel := logLevelFlag.Parse()
 			logger := logging.DefaultLogger(logLevel, formatFlag.Parse())
+			logger.Debugf("setting log-level to %s", strings.ToUpper(logLevel.String()))
```
Comment (dsu-igeek) [inline pkg/cmd/cli/upload/delete.go]: Let's change "The command, datamgr upload delete, is called " to "Datamgr upload delete called"

File: pkg/cmd/cli/upload/describe.go [hunk 5]
```diff
@@ -20,6 +21,7 @@ func NewDescribeCommand(f client.Factory, use string) *cobra.Command {
 		Run: func(c *cobra.Command, args []string) {
 			logLevel := logLevelFlag.Parse()
 			logger := logging.DefaultLogger(logLevel, formatFlag.Parse())
+			logger.Debugf("setting log-level to %s", strings.ToUpper(logLevel.String()))
 			logger.Infof("The command, datamgr upload describe, is called")
```
Comment (dsu-igeek) [inline pkg/cmd/cli/upload/describe.go]: "The command, datamgr upload describe, is called" -> "Datamgr upload describe called"

File: pkg/cmd/cli/download/get.go [hunk 6]
```diff
@@ -20,6 +21,7 @@ func NewGetCommand(f client.Factory, use string) *cobra.Command {
 		Run: func(c *cobra.Command, args []string) {
 			logLevel := logLevelFlag.Parse()
 			logger := logging.DefaultLogger(logLevel, formatFlag.Parse())
+			logger.Debugf("setting log-level to %s", strings.ToUpper(logLevel.String()))
 			logger.Infof("The command, datamgr download get, is called")
```
Comment (dsu-igeek) [inline pkg/cmd/cli/download/get.go]: Let's change "The command, datamgr download get, is called " to "Datamgr download get called"

File: pkg/cmd/cli/download/describe.go [hunk 7]
```diff
@@ -20,6 +21,7 @@ func NewDescribeCommand(f client.Factory, use string) *cobra.Command {
 		Run: func(c *cobra.Command, args []string) {
 			logLevel := logLevelFlag.Parse()
 			logger := logging.DefaultLogger(logLevel, formatFlag.Parse())
+			logger.Debugf("setting log-level to %s", strings.ToUpper(logLevel.String()))
```
Comment (dsu-igeek) [inline pkg/cmd/cli/download/describe.go]: Let's change "The command, datamgr download describe, is called " to "Datamgr download describe called"

File: pkg/dataMover/data_mover.go [hunk 8]
```diff
@@ -65,9 +68,9 @@ func (this *DataMover) CopyToRepo(peID astrolabe.ProtectedEntityID) (astrolabe.P
 		return astrolabe.ProtectedEntityID{}, err
 	}
 
-	this.Infof("Ready to call s3 PETM copy API")
+	this.Debugf("Ready to call s3 PETM copy API")
```
Comment (dsu-igeek) [inline pkg/dataMover/data_mover.go]: Please add the PEID we're copying from

File: pkg/dataMover/data_mover.go [hunk 9]
```diff
@@ -65,9 +68,9 @@ func (this *DataMover) CopyToRepo(peID astrolabe.ProtectedEntityID) (astrolabe.P
...
 	s3PE, err := this.s3PETM.Copy(ctx, updatedPE, astrolabe.AllocateNewObject)
-	this.Infof("Return from the call of s3 PETM copy API")
+	this.Debugf("Return from the call of s3 PETM copy API")
```
Comment (dsu-igeek) [inline pkg/dataMover/data_mover.go]: Please add the PEID we're copying from.  This helps if log messages get interleaved so we can know which ones are completing

File: pkg/controller/upload_controller.go [hunk 10]
```diff
@@ -197,7 +197,7 @@ func (c *uploadController) pvbHandler(obj interface{}) {
 		return
 	}
 
-	log.Debug("Enqueueing Upload")
+	log.Info("Enqueueing Upload")
```
Comment (dsu-igeek) [inline pkg/controller/upload_controller.go]: What object are we enqueueing the upload for?

File: pkg/controller/download_controller.go [hunk 11]
```diff
@@ -148,7 +148,7 @@ func (c *downloadController) processDownload(key string) error {
 				cancel()
 			},
 			OnStoppedLeading: func() {
-				log.Debug("Processed Download.")
+				log.Info("Processed Download.")
```
Comment (dsu-igeek) [inline pkg/controller/download_controller.go]: Which object did we process the download for?

File: pkg/controller/download_controller.go [hunk 12]
```diff
@@ -82,7 +82,7 @@ func (c *downloadController) enqueueDownload(obj interface{}) {
 		return
 	}
 
-	log.Debug("Enqueueing Download")
+	log.Info("Enqueueing Download")
```
Comment (dsu-igeek) [inline pkg/controller/download_controller.go]: Which object are we enqueueing the download for?

File: pkg/snapshotmgr/snapshot_manager.go [hunk 13]
```diff
@@ -152,26 +153,26 @@ func (this *SnapshotManager) CreateSnapshot(peID astrolabe.ProtectedEntityID, ta
...
-		this.Errorf("CreateSnapshot: Failed to lookup the env variable for velero namespace")
+		this.Errorf("CreateSnapshot: Failed to lookup the env variable for velero namespace: %v", err)
 		return updatedPeID, err
 	}
 
 	upload := builder.ForUpload(veleroNs, "upload-"+peSnapID.GetID()).BackupTimestamp(time.Now()).SnapshotID(updatedPeID.String()).Phase(v1api.UploadPhaseNew).Result()
 	_, err = pluginClient.VeleropluginV1().Uploads(veleroNs).Create(upload)
 	if err != nil {
-		this.Errorf("CreateSnapshot: Failed to create Upload CR")
+		this.Errorf("CreateSnapshot: Failed to create Upload CR: %v", err)
```
Comment (dsu-igeek) [inline pkg/snapshotmgr/snapshot_manager.go]: What PEID are we creating the upload CR for?

File: pkg/snapshotmgr/snapshot_manager.go [hunk 14]
```diff
@@ -230,7 +231,7 @@ func (this *SnapshotManager) DeleteProtectedEntitySnapshot(peID astrolabe.Protec
 	}
 
 	if len(snapshotIds) == 0 {
-		this.Infof("There is no snapshots from the perspective of ProtectedEntityTypeManager. Skipping the deleteSnapshot operation")
+		this.Warningf("There is no snapshots from the perspective of ProtectedEntityTypeManager. Skipping the deleteSnapshot operation")
```
Comment (dsu-igeek) [inline pkg/snapshotmgr/snapshot_manager.go]: is -> are, what PEID are we trying to delete from?

File: pkg/snapshotmgr/snapshot_manager.go [hunk 15]
```diff
@@ -221,7 +222,7 @@ func (this *SnapshotManager) DeleteProtectedEntitySnapshot(peID astrolabe.Protec
 
 	pe, err := petm.GetProtectedEntity(ctx, peID)
 	if err != nil {
-		this.Errorf("Failed to get the ProtectedEntity")
+		this.Errorf("Failed to get the ProtectedEntity: %v", err)
```
Comment (dsu-igeek) [inline pkg/snapshotmgr/snapshot_manager.go]: What is the PEID we failed to get?

File: pkg/snapshotmgr/snapshot_manager.go [hunk 16]
```diff
@@ -182,7 +183,7 @@ func (this *SnapshotManager) DeleteSnapshot(peID astrolabe.ProtectedEntityID) er
 	this.Infof("Step 1: Deleting the local snapshot, %s", peID.String())
 	err := this.DeleteProtectedEntitySnapshot(peID, false)
 	if err != nil {
-		this.Errorf("Failed to delete the local snapshot")
+		this.Errorf("Failed to delete the local snapshot: %v", err)
```
Comment (dsu-igeek) [inline pkg/snapshotmgr/snapshot_manager.go]: What PEID/snapshot ID were we unable to delete?

File: pkg/snapshotmgr/snapshot_manager.go [hunk 17]
```diff
@@ -195,7 +196,7 @@ func (this *SnapshotManager) DeleteSnapshot(peID astrolabe.ProtectedEntityID) er
 	this.Infof("Step 2: Deleting the durable snapshot, %s, from s3", peID.String())
 	err = this.DeleteProtectedEntitySnapshot(peID, true)
 	if err != nil {
-		this.Errorf("Failed to delete the durable snapshot")
+		this.Errorf("Failed to delete the durable snapshot: %v", err)
```
Comment (dsu-igeek) [inline pkg/snapshotmgr/snapshot_manager.go]: What PEID/snapshot ID were we unable to delete?

File: pkg/utils/utils.go [hunk 18]
```diff
@@ -157,7 +157,7 @@ func GetIVDPETMFromParamsMap(params map[string]interface{}, logger logrus.FieldL
 
 	ivdPETM, err := ivd.NewIVDProtectedEntityTypeManagerFromURL(&vcUrl, s3URLBase, insecure, logger)
 	if err != nil {
-		logger.Errorf("Error at creating new IVD PETM")
+		logger.Errorf("Error at creating new IVD PETM with error message %v", err)
```
Comment (dsu-igeek) [inline pkg/utils/utils.go]: Let's give the vcURL and s3URLBase here so we know what it was trying to talk to

File: pkg/controller/upload_controller.go (hunk 10 above)
Comment (lintongj) [reply pkg/controller/upload_controller.go]: The upload CR key will also be logged due the previous configuration of log fields in the same function.

`log := loggerForUpload(c.logger, req)`

File: pkg/controller/download_controller.go (hunk 12 above)
Comment (lintongj) [reply pkg/controller/download_controller.go]: The same as above.

**Comment by lintongj** [comment]
Hi Xinyan @xinyanw409, a few of general comments here.
1. Would you please follow the change in pkg/controller/upload_controller.go of https://github.com/vmware-tanzu/velero-plugin-for-vsphere/pull/22 as an example to log with fields, such as, https://github.com/vmware-tanzu/velero-plugin-for-vsphere/blob/dc61ce5a414804e2ab03d85f145b1ebb64dc6857/pkg/controller/upload_controller.go#L300. So, we don't have to explicitly log some key information, such as, volume ID/snapshot ID, upload/download CR key, etc, in every log message.
2. Please try to rebase your commit and always make sure it is on TOT(top of tree)
3. Please try to squash your commits into one single commit.

File: pkg/snapshotmgr/snapshot_manager.go (hunk 16 above)
Comment (xinyanw409) [reply pkg/snapshotmgr/snapshot_manager.go]: Add peid in the error message.

File: pkg/snapshotmgr/snapshot_manager.go (hunk 17 above)
Comment (xinyanw409) [reply pkg/snapshotmgr/snapshot_manager.go]: Same to the above.

File: Makefile (hunk 1 above)
Comment (xinyanw409) [reply Makefile]: Rebased to TOT.

File: pkg/dataMover/data_mover.go (hunk 8 above)
Comment (xinyanw409) [reply pkg/dataMover/data_mover.go]: Use log := this.WithField("Local PEID", peID.String()) with will contains the peID in each log message.

File: pkg/dataMover/data_mover.go (hunk 9 above)
Comment (xinyanw409) [reply pkg/dataMover/data_mover.go]: Same above.

File: pkg/controller/download_controller.go (hunk 11 above)
Comment (xinyanw409) [reply pkg/controller/download_controller.go]: Same above.

File: pkg/plugin/volume_snapshotter_plugin.go [hunk 19]
```diff
@@ -55,7 +55,7 @@ var _ velero.VolumeSnapshotter = (*NewVolumeSnapshotter)(nil)
 // configuration key-value pairs. It returns an error if the VolumeSnapshotter
 // cannot be initialized from the provided config. Note that after v0.10.0, this will happen multiple times.
 func (p *NewVolumeSnapshotter) Init(config map[string]string) error {
-	p.Infof("Init called", config)
+	p.Infof("Init called with config %v", config)
```
Comment (lintongj) [inline pkg/plugin/volume_snapshotter_plugin.go]: "Init called with config: %v"

File: pkg/plugin/volume_snapshotter_plugin.go [hunk 20]
```diff
@@ -122,32 +122,31 @@ const cnsBlockVolumeType = "ivd"
...
-	p.Infof("IsVolumeReady called", volumeID, volumeAZ)
+	p.Infof("IsVolumeReady called with volumeID %s and volumeAZ %s", volumeID, volumeAZ)
 	return true, nil
 }
 
 // CreateSnapshot creates a snapshot of the specified volume, and applies any provided
 // set of tags to the snapshot.
 func (p *NewVolumeSnapshotter) CreateSnapshot(volumeID, volumeAZ string, tags map[string]string) (string, error) {
-	p.Infof("CreateSnapshot called", volumeID, volumeAZ, tags)
+	p.Infof("CreateSnapshot called with volumeID %s, volumeAZ %s tags %v", volumeID, volumeAZ, tags)
```
Comment (lintongj) [inline pkg/plugin/volume_snapshotter_plugin.go]: volumeAZ %s, tags %v

Comment (xinyanw409) [reply pkg/plugin/volume_snapshotter_plugin.go]: Fixed.

File: pkg/plugin/volume_snapshotter_plugin.go (hunk 19 above)
Comment (xinyanw409) [reply pkg/plugin/volume_snapshotter_plugin.go]: fixed.

**Pull request merged**